tests/unit/fs/test_repo.py

```diff
@@ -465,17 +467,6 @@ def test_repo_fs_no_subrepos(tmp_dir, dvc, scm):
     assert set(actual) == set(expected)
     assert len(actual) == len(expected)
 
-    assert fs.isfile(tmp_dir / "lorem") is True
-    assert fs.isfile(tmp_dir / "dir" / "repo" / "foo") is False
-    assert fs.isdir(tmp_dir / "dir" / "repo") is False
-    assert fs.isdir(tmp_dir / "dir") is True
-
-    assert fs.isdvc(tmp_dir / "lorem") is True
-    assert fs.isdvc(tmp_dir / "dir" / "repo" / "dir1") is False
-
-    assert fs.exists(tmp_dir / "dir" / "repo.txt") is True
-    assert fs.exists(tmp_dir / "repo" / "ipsum") is False
-
```

Comment:
Hm, this is actually what we were talking about in https://github.com/iterative/dvc/pull/5812#discussion_r629539919 . Here we have a very strange situation where `fs.walk` is dvcignore aware, but `isdir/isdfile` are not, which is bad.  So we need to either use dvcignore in both, or just apply dvcignore where it is needed on top of RepoFileSystem, effectively decoupling it and dvcignore.

Reply:
Simple fix for now is to just use `dvicgnore` in `RepoFileSystem`s `isfile/isdir/exists`, but we might want to think about futher decoupling here in the future.